rescript/get_unite_data.py

```diff
@@ -0,0 +1,167 @@
+    print('Get URLs for these DOIs:', doi)
+    base_url = 'https://api.plutof.ut.ee/v1/public/dois/'\
+               '?format=vnd.api%2Bjson&identifier='
+    query_data = requests.get(base_url + doi).json()
+    # Updates can be made to files in a DOI, so on the advice of the devs,
+    # only return the last (newest) file with this -1  vv
+    URL = query_data['data'][0]['attributes']['media'][-1]['url']
+    return URL
+
+# Testing
+_unite_get_url(doi = '10.15156/BIO/2938079')
+
+# with tempfile.TemporaryDirectory() as tmp_dir:
+tmp_dir = tempfile.mkdtemp()
+
+
+def _unite_get_tgz(url, download_path):
+    print('Downloading ' + url)
+    response = requests.get(url, stream=True)
+    if response.status_code != 200:
+        raise ValueError("Failed to download the file from " + url)
+    # Save .tgz file
+    tar_file_path = os.path.join(download_path, 'unitefile.tar.gz')
+    with open(tar_file_path, 'wb') as f:
+        f.write(response.content)
+    # Extract only the 'developer' subdirectory
+    with tarfile.open(tar_file_path, 'r:gz') as tar:
+        # Ensure that 'developer' exists in the tar file
+        members = [member for member in tar.getmembers() if member.name.startswith('developer')]
+        if not members:
+            raise ValueError("No 'developer' subdirectory found in the .tar.gz file.")
+        for member in members:
+            member.name = os.path.basename(member.name) # Strip the 'developer' prefix
+            tar.extract(member, path=download_path)
+    
+    return download_path
+
+# Test it by downloading this file
+_unite_get_tgz(_unite_get_url(doi = '10.15156/BIO/2938079'), tmp_dir)
+
+
+def get_unite_data(version, taxon_group, singletons=False):
+    doi = _unite_get_doi(version, taxon_group, singletons)
+    url = _unite_get_url(doi)
+    # Eventual output
+    results = {'sequences': [], 'taxonomy': []}
+    # open temp output files, somehow?
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # tmp_dir = tempfile.mkdtemp()
+        print('Temporary directory:', tmpdirname)
+        _unite_get_tgz(url, download_path=tmpdirname)
+        # Find and import artifacts based on suffix
+        for root, dirs, files in os.walk(tmpdirname):
+            for file in files:
+                print(results)
+                if file.endswith('.fasta'):
+                    fasta_file_name = os.path.join(root, file)
+                    print('found fasta: ' + fasta_file_name)
+                    results['sequences'].append(qiime2.Artifact.import_data('FeatureData[Sequence]', fasta_file_name))
+                elif file.endswith('.txt'):
+                    txt_file_name = os.path.join(root, file)
+                    print('found txt: ' + txt_file_name)
+                    results['taxonomy'].append(qiime2.Artifact.import_data('FeatureData[Taxonomy]', txt_file_name))
+    return results
+
+# Testing
+example_output = get_unite_data(version='9.0', taxon_group='fungi')
```

Comment:
It is not clear to me how you can disambiguate the different FASTA and Taxonomy files from one another. So that the user can pick which one they want to use. It might be best to pull the information from the file name as you're walking through and use that to later "hard-code" a generic name for each of these output files, perhaps set up as a tuple or dict? Like this:

```
example_output = {
    'sequences' : 
        ('sh_refs_qiime_ver9_97_25.07.2023_dev',  
         <artifact: FeatureData[Sequence] uuid: 3f602392-6416-4742-9710-fd2778a95227>),
        ('sh_refs_qiime_ver9_99_25.07.2023_dev',
         <artifact: FeatureData[Sequence] uuid: 773ac252-3c04-42b4-a5bc-74d13fe9a629>),
         ...,
    'taxonomy':
        ('sh_taxonomy_qiime_ver9_97_25.07.2023_dev', 
         <artifact: FeatureData[Taxonomy] uuid: f39464a8-b3a5-47f0-bb8c-9889c9da6431>),
        ('sh_taxonomy_qiime_ver9_99_25.07.2023_dev',
          <artifact: FeatureData[Taxonomy] uuid: e8bd1e45-c311-4730-ac4f-126bd39ca50e>),
         ...}
                                                      ]}
```
Then later here, and in `plugin_setup.py` you'd have explicit file names for each output file.

Reply:
Good idea! I don't capture database cluster %id anywhere, as I can only see if once the .tgz file is downloaded...

If for any reason these files are not in the same order, that could also break things.

Reply:
> If for any reason these files are not in the same order, that could also break things.

That is why I prefer being explicit here. You never know what'll happen. 😄